### PR TITLE
When publishing on a branch, don't tag as latest

### PR DESCRIPTION
**Pull request by jessitron**
This uses sdm#404 to tag npm publishment as "branch" when it isn't on master; this has the effect of NOT tagging it as "latest" so that the latest one will be on master.

Currently when you push to sdm (even on a branch) it publishes _everything_ as "latest"

### CODE REVIEW

**Comment by jessitron** [comment]
alternatively, perhaps we want to tag as "local" (or anything really, that's not "latest") anything that is not a release?

**Comment by cdupuis** [comment]
What about tagging it with the branch name?

File: src/machine/nodeSupport.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ import { tslintFix } from "@atomist/sdm/pack/node/tslint";
 import { tagRepo } from "@atomist/sdm/util/github/tagRepo";
 import { AutomationClientTagger } from "../support/tagger";
 import {
-    ProductionDeploymentGoal,
+    ProductionDeploymentGoal, PublishBranchGoal,
```
Comment (ddgenome) [inline src/machine/nodeSupport.ts]: Inconsistent formatting.

File: src/machine/nodeSupport.ts [hunk 2]
```diff
@@ -111,6 +111,14 @@ export const NodeSupport: ExtensionPack = {
                     {
                         ...sdm.configuration.sdm.npm as NpmOptions,
                     }), { pushTest: IsNode })
+            .addGoalImplementation("nodePublishBranch", PublishBranchGoal,
+                executePublish(sdm.configuration.sdm.projectLoader,
+                    NodeProjectIdentifier,
+                    NpmPreparations,
+                    {
+                        ...sdm.configuration.sdm.npm as NpmOptions,
+                        tag: "branch",
```
Comment (ddgenome) [inline src/machine/nodeSupport.ts]: It would be good to use the branch name for the tag.

**Comment by jessitron** [comment]
Neither of these are possible while using ‘executePublish’ ... but we could wrap it or (better?) change it to do this as standard

**Comment by ddgenome** [comment]
So the issue is that there is no way to persist the tag in the preparations such that it would be picked up by `executePublish`?

**Comment by jessitron** [comment]
replaced by https://github.com/atomist/sdm/pull/406

**Pull request closed without merging**